### ISSUE TITLE
[ClusterInfraAssessmentReport] failed to create ClusterPolicyReport: ClusterPolicyReport.wgpolicyk8s.io

### ISSUE THREAD

**Issue by sherifkayad** [fjogeleit/trivy-operator-polr-adapter#118]
Hello 👋 

With the latest version of both the POLR Adapter and Trivy, we are currently getting:
```
2023/07/27 12:06:53 [ERROR] ClusterInfraAssessmentReport: Failed to process report node-ip-10-4-34-125.eu-central-1.compute.internal; failed to create ClusterPolicyReport: ClusterPolicyReport.wgpolicyk8s.io "trivy-infra-cpolr-node-ip-10-4-34-125.eu-central-1.compute.internal" is invalid: [metadata.ownerReferences.apiVersion: Invalid value: "": version must not be empty, metadata.ownerReferences.kind: Invalid value: "": kind must not be empty]
```

### My current versions
- Trivy POLR Adapter => `0.6.0`
- Trivy Operator => `0.15.1`
- Note sure if it's relevant, however, Kubernetes version => `1.27` on AWS EKS

### A sample of my Trivy ClusterInfraAssessmentReport CRD
<details>
  <summary>Click to expand</summary>
 
  ```yaml
  apiVersion: aquasecurity.github.io/v1alpha1
  kind: ClusterInfraAssessmentReport
  metadata:
    creationTimestamp: '2023-07-27T12:06:52Z'
    generation: 1
    labels:
      plugin-config-hash: 659b7b9c46
      resource-spec-hash: 767485d8b5
      trivy-operator.resource.kind: Node
      trivy-operator.resource.name: ip-10-4-34-125.eu-central-1.compute.internal
      trivy-operator.resource.namespace: ''
    managedFields:
      - apiVersion: aquasecurity.github.io/v1alpha1
        fieldsType: FieldsV1
        fieldsV1:
          f:metadata:
            f:labels:
              .: {}
              f:plugin-config-hash: {}
              f:resource-spec-hash: {}
              f:trivy-operator.resource.kind: {}
              f:trivy-operator.resource.name: {}
              f:trivy-operator.resource.namespace: {}
            f:ownerReferences:
              .: {}
              k:{"uid":"0589c43a-b7e8-4416-a28c-2dcbadd49f1d"}: {}
          f:report:
            .: {}
            f:checks: {}
            f:scanner:
              .: {}
              f:name: {}
              f:vendor: {}
              f:version: {}
            f:summary:
              .: {}
              f:criticalCount: {}
              f:highCount: {}
              f:lowCount: {}
              f:mediumCount: {}
        manager: Go-http-client
        operation: Update
        time: '2023-07-27T12:06:52Z'
    name: node-ip-10-4-34-125.eu-central-1.compute.internal
    ownerReferences:
      - apiVersion: v1
        blockOwnerDeletion: false
        controller: true
        kind: Node
        name: ip-10-4-34-125.eu-central-1.compute.internal
        uid: 0589c43a-b7e8-4416-a28c-2dcbadd49f1d
    resourceVersion: '180364507'
    uid: 7d56069b-012e-45a3-9510-6ad315d9c1b1
    selfLink: >-
      /apis/aquasecurity.github.io/v1alpha1/clusterinfraassessmentreports/node-ip-10-4-34-125.eu-central-1.compute.internal
  report:
    checks:
      - category: Kubernetes Security Check
        checkID: KCV0047
        description: Do not use self-signed certificates for TLS.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --peer-auto-tls argument is not set to true
      - category: Kubernetes Security Check
        checkID: KCV0089
        description: Setup TLS connection on the Kubelets.
        messages:
          - Ensure that the --tls-key-file argument are set as appropriate
        severity: CRITICAL
        success: false
        title: Ensure that the --tls-key-file argument are set as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0135
        description: Use individual service account credentials for each controller.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --use-service-account-credentials argument is set to
          true
      - category: Kubernetes Security Check
        checkID: KCV0085
        description: Do not disable timeouts on streaming connections.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the --streaming-connection-idle-timeout argument is not set
          to 0
      - category: Kubernetes Security Check
        checkID: KCV0079
        description: Disable anonymous requests to the Kubelet server.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: Ensure that the --anonymous-auth argument is set to false
      - category: Kubernetes Security Check
        checkID: KCV0037
        description: >-
          Allow pods to verify the API server's serving certificate before
          establishing connections.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --root-ca-file argument is set as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0042
        description: Configure TLS encryption for the etcd service.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --cert-file and --key-file arguments are set as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0001
        description: Disable anonymous requests to the API server.
        messages:
          - ''
        severity: MEDIUM
        success: true
        title: Ensure that the --anonymous-auth argument is set to false
      - category: Kubernetes Security Check
        checkID: KCV0092
        description: >-
          Ensure that the Kubelet is configured to only use strong cryptographic
          ciphers.
        messages:
          - Ensure that the Kubelet only makes use of Strong Cryptographic Ciphers
        severity: CRITICAL
        success: false
        title: Ensure that the Kubelet only makes use of Strong Cryptographic Ciphers
      - category: Kubernetes Security Check
        checkID: KCV0087
        description: >-
          Security relevant information should be captured. The --event-qps flag
          on the Kubelet can be used to limit the rate at which events are
          gathered
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the --event-qps argument is set to 0 or a level which
          ensures appropriate event capture
      - category: Kubernetes Security Check
        checkID: KCV0033
        description: Activate garbage collector on pod termination, as appropriate.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --terminated-pod-gc-threshold argument is set as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0006
        description: Verify kubelet's certificate before establishing connection.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --kubelet-certificate-authority argument is set as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0019
        description: >-
          Enable auditing on the Kubernetes API Server and set the desired audit
          log path.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --audit-log-path argument is set
      - category: Kubernetes Security Check
        checkID: KCV0071
        description: >-
          If kube-proxy is running, and if it is using a file-based kubeconfig
          file, ensure that the proxy kubeconfig file has permissions of 600 or
          more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          If proxy kubeconfig file exists ensure permissions are set to 600 or
          more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0077
        description: >-
          Ensure that if the kubelet refers to a configuration file with the
          --config argument, that file has permissions of 600 or more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          If the kubelet config.yaml configuration file is being used validate
          permissions set to 600 or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0063
        description: Ensure that the scheduler config  file ownership is set to root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: Ensure that the scheduler config  file ownership is set to root:root
      - category: Kubernetes Security Check
        checkID: KCV0048
        description: >-
          Ensure that the API server pod specification file has permissions of 600
          or more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the API server pod specification file permissions are set to
          600 or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0069
        description: >-
          Ensure that the kubelet service file has permissions of 600 or more
          restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the kubelet service file permissions are set to 600 or more
          restrictive
      - category: Kubernetes Security Check
        checkID: KCV0065
        description: >-
          Ensure that the controller-manager config  file ownership is set to
          root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the controller-manager config  file ownership is set to
          root:root
      - category: Kubernetes Security Check
        checkID: KCV0057
        description: >-
          Ensure that the container network interface file ownership is set to
          root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the container network interface file ownership is set to
          root:root
      - category: Kubernetes Security Check
        checkID: KCV0026
        description: >-
          etcd should be configured to make use of TLS encryption for client
          connections.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --etcd-certfile and --etcd-keyfile arguments are set as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0007
        description: Do not always authorize all requests.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --authorization-mode argument is not set to AlwaysAllow
      - category: Kubernetes Security Check
        checkID: KCV0036
        description: >-
          Explicitly set a service account private key file for service accounts
          on the controller manager.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --service-account-private-key-file argument is set as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0078
        description: >-
          Ensure that if the kubelet refers to a configuration file with the
          --config argument, that file is owned by root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          If the kubelet config.yaml configuration file is being used validate
          file ownership is set to root:root 
      - category: Kubernetes Security Check
        checkID: KCV0052
        description: >-
          Ensure that the scheduler pod specification file has permissions of 600
          or more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the scheduler pod specification file permissions are set to
          600 or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0013
        description: >-
          The SecurityContextDeny admission controller can be used to deny pods
          which make use of some SecurityContext fields which could allow for
          privilege escalation in the cluster. This should be used where
          PodSecurityPolicy is not in place within the cluster.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the admission control plugin SecurityContextDeny is set if
          PodSecurityPolicy is not used
      - category: Kubernetes Security Check
        checkID: KCV0003
        description: >-
          This admission controller rejects all net-new usage of the Service field
          externalIPs.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --DenyServiceExternalIPs is not set
      - category: Kubernetes Security Check
        checkID: KCV0014
        description: Automate service accounts management.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the admission control plugin ServiceAccount is set
      - category: Kubernetes Security Check
        checkID: KCV0084
        description: Allow Kubelet to manage iptables.
        messages:
          - ''
        severity: HIGH
        success: true
        title: Ensure that the --make-iptables-util-chains argument is set to true
      - category: Kubernetes Security Check
        checkID: KCV0067
        description: Ensure that the Kubernetes PKI key file permission is set to 600.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: Ensure that the Kubernetes PKI key file permission is set to 600
      - category: Kubernetes Security Check
        checkID: KCV0088
        description: Setup TLS connection on the Kubelets.
        messages:
          - Ensure that the --tls-cert-file argument are set as appropriate
        severity: CRITICAL
        success: false
        title: Ensure that the --tls-cert-file argument are set as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0011
        description: Do not allow all requests.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the admission control plugin AlwaysAdmit is not set
      - category: Kubernetes Security Check
        checkID: KCV0081
        description: Enable Kubelet authentication using certificates.
        messages:
          - Ensure that the --client-ca-file argument is set as appropriate
        severity: CRITICAL
        success: false
        title: Ensure that the --client-ca-file argument is set as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0004
        description: Use https for kubelet connections.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --kubelet-https argument is set to true
      - category: Kubernetes Security Check
        checkID: KCV0044
        description: Do not use self-signed certificates for TLS.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --auto-tls argument is not set to true
      - category: Kubernetes Security Check
        checkID: KCV0091
        description: Enable kubelet server certificate rotation.
        messages:
          - Verify that the RotateKubeletServerCertificate argument is set to true
        severity: HIGH
        success: false
        title: Verify that the RotateKubeletServerCertificate argument is set to true
      - category: Kubernetes Security Check
        checkID: KCV0056
        description: >-
          Ensure that the container network interface file has permissions of 600
          or more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the container network interface file permissions are set to
          600 or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0017
        description: Do not disable the secure port.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --secure-port argument is not set to 0
      - category: Kubernetes Security Check
        checkID: KCV0061
        description: Ensure that the admin config  file ownership is set to root:root.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: Ensure that the admin config  file ownership is set to root:root
      - category: Kubernetes Security Check
        checkID: KCV0028
        description: Setup TLS connection on the API server.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --client-ca-file argument is set as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0040
        description: Disable profiling, if not needed.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --profiling argument is set to false
      - category: Kubernetes Security Check
        checkID: KCV0002
        description: Do not use token based authentication.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --token-auth-file parameter is not set
      - category: Kubernetes Security Check
        checkID: KCV0054
        description: >-
          Ensure that the etcd pod specification file has permissions of 600 or
          more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the etcd pod specification file permissions are set to 600
          or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0070
        description: Ensure that the kubelet service file ownership is set to root:root.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: Ensure that the kubelet service file ownership is set to root:root
      - category: Kubernetes Security Check
        checkID: KCV0021
        description: Retain 10 or an appropriate number of old log files.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --audit-log-maxbackup argument is set to 10 or as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0045
        description: >-
          etcd should be configured to make use of TLS encryption for peer
          connections.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --peer-cert-file and --peer-key-file arguments are set
          as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0015
        description: Reject creating objects in a namespace that is undergoing termination.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the admission control plugin NamespaceLifecycle is set
      - category: Kubernetes Security Check
        checkID: KCV0075
        description: >-
          Ensure that the certificate authorities file has permissions of 600 or
          more restrictive.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: >-
          Ensure that the certificate authorities file permissions are set to 600
          or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0080
        description: Do not allow all requests. Enable explicit authorization.
        messages:
          - >-
            Ensure that the --authorization-mode argument is not set to
            AlwaysAllow
        severity: HIGH
        success: false
        title: Ensure that the --authorization-mode argument is not set to AlwaysAllow
      - category: Kubernetes Security Check
        checkID: KCV0055
        description: >-
          Ensure that the etcd pod specification file ownership is set to
          root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the etcd pod specification file ownership is set to
          root:root
      - category: Kubernetes Security Check
        checkID: KCV0068
        description: >-
          Ensure that the Kubernetes PKI certificate file permission is set to
          600.
        messages:
          - ''
        severity: HIGH
        success: true
        title: Ensure that the Kubernetes PKI certificate file permission is set to 600
      - category: Kubernetes Security Check
        checkID: KCV0038
        description: Enable kubelet server certificate rotation on controller-manager.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the RotateKubeletServerCertificate argument is set to true
      - category: Kubernetes Security Check
        checkID: KCV0043
        description: Enable client authentication on etcd service.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --client-cert-auth argument is set to true
      - category: Kubernetes Security Check
        checkID: KCV0029
        description: >-
          etcd should be configured to make use of TLS encryption for client
          connections.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --etcd-cafile argument is set as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0066
        description: >-
          Ensure that the Kubernetes PKI directory and file file ownership is set
          to root:root.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: >-
          Ensure that the Kubernetes PKI directory and file file ownership is set
          to root:root
      - category: Kubernetes Security Check
        checkID: KCV0046
        description: etcd should be configured for peer authentication.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --peer-client-cert-auth argument is set to true
      - category: Kubernetes Security Check
        checkID: KCV0008
        description: Restrict kubelet nodes to reading only objects associated with them.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --authorization-mode argument includes Node
      - category: Kubernetes Security Check
        checkID: KCV0025
        description: >-
          Explicitly set a service account public key file for service accounts on
          the apiserver.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --service-account-key-file argument is set as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0027
        description: Setup TLS connection on the API server.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --tls-cert-file and --tls-private-key-file arguments are
          set as appropriate
      - category: Kubernetes Security Check
        checkID: KCV0009
        description: Turn on Role Based Access Control.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --authorization-mode argument includes RBAC
      - category: Kubernetes Security Check
        checkID: KCV0022
        description: Rotate log files on reaching 100 MB or as appropriate.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --audit-log-maxsize argument is set to 100 or as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0020
        description: Retain the logs for at least 30 days or as appropriate.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --audit-log-maxage argument is set to 30 or as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0083
        description: >-
          Protect tuned kernel parameters from overriding kubelet default kernel
          parameter values.
        messages:
          - Ensure that the --protect-kernel-defaults is set to true
        severity: HIGH
        success: false
        title: Ensure that the --protect-kernel-defaults is set to true
      - category: Kubernetes Security Check
        checkID: KCV0074
        description: Ensure that the kubelet.conf file ownership is set to root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the --kubeconfig kubelet.conf file ownership is set to
          root:root
      - category: Kubernetes Security Check
        checkID: KCV0082
        description: Disable the read-only port.
        messages:
          - ''
        severity: HIGH
        success: true
        title: Verify that the --read-only-port argument is set to 0
      - category: Kubernetes Security Check
        checkID: KCV0049
        description: >-
          Ensure that the API server pod specification file ownership is set to
          root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the API server pod specification file ownership is set to
          root:root
      - category: Kubernetes Security Check
        checkID: KCV0064
        description: >-
          Ensure that the controller-manager config file has permissions of 600 or
          more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the controller-manager config file permissions are set to
          600 or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0016
        description: Limit the Node and Pod objects that a kubelet could modify.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the admission control plugin NodeRestriction is set
      - category: Kubernetes Security Check
        checkID: KCV0062
        description: >-
          Ensure that the scheduler config file has permissions of 600 or more
          restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the scheduler config file permissions are set to 600 or more
          restrictive
      - category: Kubernetes Security Check
        checkID: KCV0018
        description: Disable profiling, if not needed.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --profiling argument is set to false
      - category: Kubernetes Security Check
        checkID: KCV0060
        description: >-
          Ensure that the admin config file has permissions of 600 or more
          restrictive.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: >-
          Ensure that the admin config file permissions are set to 600 or more
          restrictive
      - category: Kubernetes Security Check
        checkID: KCV0073
        description: >-
          Ensure that the kubelet.conf file has permissions of 600 or more
          restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the --kubeconfig kubelet.conf file permissions are set to
          600 or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0051
        description: >-
          Ensure that the controller manager pod specification file ownership is
          set to root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the controller manager pod specification file ownership is
          set to root:root
      - category: Kubernetes Security Check
        checkID: KCV0059
        description: Ensure that the etcd data directory ownership is set to etcd:etcd.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the etcd data directory ownership is set to etcd:etcd
      - category: Kubernetes Security Check
        checkID: KCV0050
        description: >-
          Ensure that the controller manager pod specification file has
          permissions of 600 or more restrictive.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the controller manager pod specification file permissions
          are set to 600 or more restrictive
      - category: Kubernetes Security Check
        checkID: KCV0072
        description: >-
          If kube-proxy is running, ensure that the file ownership of its
          kubeconfig file is set to root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: if proxy kubeconfig file exists ensure ownership is set to root:root
      - category: Kubernetes Security Check
        checkID: KCV0039
        description: Do not bind the scheduler service to non-loopback insecure addresses.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --bind-address argument is set to 127.0.0.1
      - category: Kubernetes Security Check
        checkID: KCV0024
        description: Validate service account before validating token.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --service-account-lookup argument is set to true
      - category: Kubernetes Security Check
        checkID: KCV0010
        description: Limit the rate at which the API server accepts requests.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the admission control plugin EventRateLimit is set
      - category: Kubernetes Security Check
        checkID: KCV0086
        description: Do not override node hostnames.
        messages:
          - Ensure that the --hostname-override argument is not set
        severity: HIGH
        success: false
        title: Ensure that the --hostname-override argument is not set
      - category: Kubernetes Security Check
        checkID: KCV0030
        description: Encrypt etcd key-value store.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --encryption-provider-config argument is set as
          appropriate
      - category: Kubernetes Security Check
        checkID: KCV0058
        description: >-
          Ensure that the etcd data directory has permissions of 700 or more
          restrictive.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the etcd data directory permissions are set to 700 or more
          restrictive
      - category: Kubernetes Security Check
        checkID: KCV0090
        description: Enable kubelet client certificate rotation.
        messages:
          - Ensure that the --rotate-certificates argument is not set to false
        severity: HIGH
        success: false
        title: Ensure that the --rotate-certificates argument is not set to false
      - category: Kubernetes Security Check
        checkID: KCV0053
        description: >-
          Ensure that the scheduler pod specification file ownership is set to
          root:root.
        messages:
          - ''
        severity: HIGH
        success: true
        title: >-
          Ensure that the scheduler pod specification file ownership is set to
          root:root
      - category: Kubernetes Security Check
        checkID: KCV0076
        description: >-
          Ensure that the certificate authorities file ownership is set to
          root:root.
        messages:
          - ''
        severity: CRITICAL
        success: true
        title: >-
          Ensure that the client certificate authorities file ownership is set to
          root:root
      - category: Kubernetes Security Check
        checkID: KCV0034
        description: Disable profiling, if not needed.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --profiling argument is set to false
      - category: Kubernetes Security Check
        checkID: KCV0041
        description: Do not bind the scheduler service to non-loopback insecure addresses.
        messages:
          - ''
        severity: LOW
        success: true
        title: Ensure that the --bind-address argument is set to 127.0.0.1
      - category: Kubernetes Security Check
        checkID: KCV0005
        description: Enable certificate based kubelet authentication.
        messages:
          - ''
        severity: LOW
        success: true
        title: >-
          Ensure that the --kubelet-client-certificate and --kubelet-client-key
          arguments are set as appropriate
    scanner:
      name: Trivy
      vendor: Aqua Security
      version: 0.15.0
    summary:
      criticalCount: 4
      highCount: 5
      lowCount: 0
      mediumCount: 0
  ```
</details>

**Comment by fjogeleit** [comment]
hey, thanks for reporting. I will have a look

**Comment by sherifkayad** [comment]
awesome! Thanks a lot

**Comment by sherifkayad** [comment]
@fjogeleit I also noticed something else in the logs:
```
2023/07/27 12:15:36 [ERROR] ClusterInfraAssessmentReport: Failed to delete report node-ip-10-4-42-36.eu-central-1.compute.internal; an empty namespace may not be set when a resource name is provided
```

Not sure if that's related ..

**Comment by sherifkayad** [comment]
@fjogeleit I am seeing now a panic `panic: runtime error: invalid memory address or nil pointer dereference` .. I am on Kubernetes 1.27 .. have you already upgraded the k8s API Go Lib? (https://github.com/kubernetes/kubernetes/issues/118361#issuecomment-1579198407)

**Comment by fjogeleit** [comment]
Hm can you post the panic message? Do you see some stacktraces? Will have a deeper look, sorry.

**Comment by sherifkayad** [comment]
All good thanks for your prompt answer. Here you go with the whole log dump:
```
[INFO] start server on port 8080
[INFO] ConfigAuditReports enabled
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x14e6f8a]

goroutine 1 [running]:
github.com/fjogeleit/trivy-operator-polr-adapter/pkg/adapters/auditr.(*Client).StartWatching(0xc0002dfda0, {0x1ae16c8?, 0xc000138000})
	/app/pkg/adapters/auditr/client.go:25 +0x4a
github.com/fjogeleit/trivy-operator-polr-adapter/cmd.newRunCMD.func1(0xc0004a8300, {0x1853b41?, 0x0?, 0x0?})
	/app/cmd/run.go:73 +0x285
github.com/spf13/cobra.(*Command).execute(0xc0004a8300, {0x26c2880, 0x0, 0x0})
	/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004a8000)
	/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
main.main()
	/app/main.go:13 +0x2c
```

**Comment by fjogeleit** [comment]
Thanks, I fixed the logger issue and tested it local. I released a version `0.7.2`. You can test it again after the build finished.

**Comment by sherifkayad** [comment]
Unfortunately still the same error 😢

**Comment by fjogeleit** [comment]
The panic or the APIVersion issue?

**Comment by sherifkayad** [comment]
The panic I meant ..

![Screenshot 2023-07-27 185106](https://github.com/fjogeleit/trivy-operator-polr-adapter/assets/8362784/10bcf02b-83d8-4cde-8d42-0d3fc0bd0466)

**Comment by fjogeleit** [comment]
Hm okay, will check again

**Comment by sherifkayad** [comment]
No stress .. currently I rolled back to `0.6.0` where the POLR Adapter isn't panicking ..

**Comment by fjogeleit** [comment]
Hm audit report was not enabled in my test. Tested now all reports and published a `0.7.3`. Sorry for the issues.

```
[INFO] start server on port 8080
[INFO] ConfigAuditReports enabled
[INFO] VulnerabilityReports enabled
[INFO] ComplianceReports enabled
[INFO] RbacAssessmentReports enabled
[INFO] ExposedSecretReports enabled
[INFO] InfraAssessmentReportClient enabled
[INFO] ClusterInfraAssessmentReportClient enabled
I0727 18:59:10.680466   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="configaudit" "source"="kind source: *v1alpha1.ConfigAuditReport"
I0727 18:59:10.680767   27703 controller.go:185]  "msg"="Starting Controller" "controller"="configaudit"
I0727 18:59:10.680773   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="exposedsecret" "source"="kind source: *v1alpha1.ExposedSecretReport"
I0727 18:59:10.680795   27703 controller.go:185]  "msg"="Starting Controller" "controller"="exposedsecret"
I0727 18:59:10.680467   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="compliance" "source"="kind source: *v1alpha1.ClusterComplianceReport"
I0727 18:59:10.680842   27703 controller.go:185]  "msg"="Starting Controller" "controller"="compliance"
I0727 18:59:10.680484   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="vulnerability" "source"="kind source: *v1alpha1.VulnerabilityReport"
I0727 18:59:10.680887   27703 controller.go:185]  "msg"="Starting Controller" "controller"="vulnerability"
I0727 18:59:10.680492   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="clusterrbacassessment" "source"="kind source: *v1alpha1.ClusterRbacAssessmentReport"
I0727 18:59:10.680634   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="infraassessment" "source"="kind source: *v1alpha1.InfraAssessmentReport"
I0727 18:59:10.681103   27703 controller.go:185]  "msg"="Starting Controller" "controller"="infraassessment"
I0727 18:59:10.680866   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="clusterinfraassessment" "source"="kind source: *v1alpha1.ClusterInfraAssessmentReport"
I0727 18:59:10.681174   27703 controller.go:185]  "msg"="Starting Controller" "controller"="clusterinfraassessment"
I0727 18:59:10.680921   27703 controller.go:177]  "msg"="Starting EventSource" "controller"="rbacassessment" "source"="kind source: *v1alpha1.RbacAssessmentReport"
I0727 18:59:10.680970   27703 controller.go:185]  "msg"="Starting Controller" "controller"="clusterrbacassessment"
I0727 18:59:10.681313   27703 controller.go:185]  "msg"="Starting Controller" "controller"="rbacassessment"
I0727 18:59:10.818262   27703 controller.go:219]  "msg"="Starting workers" "controller"="clusterrbacassessment" "worker count"=1
I0727 18:59:10.818338   27703 controller.go:219]  "msg"="Starting workers" "controller"="vulnerability" "worker count"=1
I0727 18:59:10.818343   27703 controller.go:219]  "msg"="Starting workers" "controller"="infraassessment" "worker count"=1
I0727 18:59:10.818396   27703 controller.go:219]  "msg"="Starting workers" "controller"="clusterinfraassessment" "worker count"=1
I0727 18:59:10.818477   27703 controller.go:219]  "msg"="Starting workers" "controller"="configaudit" "worker count"=1
I0727 18:59:10.818928   27703 controller.go:219]  "msg"="Starting workers" "controller"="rbacassessment" "worker count"=1
I0727 18:59:10.819273   27703 controller.go:219]  "msg"="Starting workers" "controller"="exposedsecret" "worker count"=1
I0727 18:59:10.819403   27703 controller.go:219]  "msg"="Starting workers" "controller"="compliance" "worker count"=1
```

**Comment by fjogeleit** [comment]
I released Helm Chart version `0.7.0` with the fixed image version `0.7.3`
